plugins/kubernetes/app/models/kuber_deploy_service.rb

```diff
@@ -60,4 +60,11 @@ def log(msg, extra_info = {})
 
     Kubernetes::Util.log msg, extra_info
   end
+
+  # Starts a new deploy watcher, forcing it to get in synch with the cluster
+  def start_watching_cluster
+    watcher = Celluloid::Actor[project.deploy_watcher_actor_name]
+    watcher.terminate if watcher and watcher.alive?
+    Celluloid::Actor[project.deploy_watcher_actor_name] = Watchers::DeployWatcher.new(project)
```

Comment:
I gave up on my intention of having this DeployWatchers always active, because it would mean a live and running instance of the watcher per project. In my opinion, that's a big overhead for Samson, but that's something I wanted to discuss with you and @henders.

If we decide to go along that path then, yes, I think this could become a supervised Actor that we just need to restart each time a new Release is created (to force it to get in-sync with the Cluster).